src/ota/interface/OTAInterfaceDefault.cpp

```diff
@@ -94,19 +93,26 @@ OTACloudProcessInterface::State OTADefaultCloudProcessInterface::fetch() {
   uint32_t start = millis();
 
   do {
-    if(http_client->available() == 0) {
-      goto exit;
-    }
+    if(http_client->available()) {
+      http_res = http_client->read(context->buffer, context->buf_len);
+
+      if(http_res < 0) {
+        DEBUG_VERBOSE("OTA ERROR: Download read error %d", http_res);
+        res = OtaDownloadFail;
+        goto exit;
+      }
 
-    http_res = http_client->read(context->buffer, context->buf_len);
+      parseOta(context->buffer, http_res);
 
-    if(http_res < 0) {
-      DEBUG_VERBOSE("OTA ERROR: Download read error %d", http_res);
-      res = OtaDownloadFail;
-      goto exit;
+      if(context->writeError) {
+        DEBUG_VERBOSE("OTA ERROR: File write error");
+        res = ErrorWriteUpdateFileFail;
+        goto exit;
+      }
+    } else {
+      /* Avoid tight loop and allow yield */
+      delay(1);
```

Comment:
This change is a consequence of the previous comment

```suggestion
    if(context->writeError) {
      DEBUG_VERBOSE("OTA ERROR: File write error");
      res = ErrorWriteUpdateFileFail;
      goto exit;
    }
```